src/vcpkg/commands.update-baseline.cpp

```diff
@@ -0,0 +1,146 @@
+}
+
+namespace vcpkg::Commands
+{
+    static const CommandStructure COMMAND_STRUCTURE{
+        create_example_string("x-update-baseline"),
+        0,
+        0,
+    };
+
+    static constexpr StringLiteral builtin_git_url = "https://github.com/microsoft/vcpkg";
+
+    static Optional<std::string> get_baseline_from_git_repo(const VcpkgPaths& paths, StringView url)
+    {
+        auto res = paths.git_fetch_from_remote_registry(url, "HEAD");
+        if (res.has_value())
+        {
+            return std::move(*res.get());
+        }
+        else
+        {
+            msg::println(Color::warning, msgUpdateBaselineGitError, msg::url = url);
+            msg::write_unlocalized_text_to_stdout(Color::warning, res.error());
+            return nullopt;
+        }
+    }
+
+    static Optional<std::string> get_latest_baseline(const VcpkgPaths& paths, const RegistryConfig& r)
+    {
+        if (r.kind == "git")
+        {
+            return get_baseline_from_git_repo(paths, r.repo.value_or_exit(VCPKG_LINE_INFO));
+        }
+        else if (r.kind == "builtin")
+        {
+            return get_baseline_from_git_repo(paths, builtin_git_url);
+        }
+        else
+        {
+            return nullopt;
+        }
+    }
+
+    void UpdateBaselineCommand::perform_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths& paths) const
+    {
+        args.parse_arguments(COMMAND_STRUCTURE);
+
+        bool has_builtin_baseline = false;
+        bool config_in_manifest = false;
+        vcpkg::Configuration configuration = paths.get_configuration();
+
+        vcpkg::Path configuration_file;
+
+        auto opt_config_path = paths.get_configuration_path();
+        if (auto p_config_path = opt_config_path.get())
+        {
+            configuration_file = std::move(*p_config_path);
+        }
+
+        Json::Object manifest;
+        vcpkg::Path manifest_file;
+        auto opt_manifest = paths.get_manifest();
+        if (auto p_manifest = opt_manifest.get())
+        {
+            manifest = *p_manifest;
+            // paths.get_manifest().has_value() == paths.get_manifest_path().has_value()
+            manifest_file = paths.get_manifest_path().value_or_exit(VCPKG_LINE_INFO);
+            if (p_manifest->contains("builtin-baseline"))
+            {
+                has_builtin_baseline = true;
+            }
+            if (p_manifest->contains("vcpkg-configuration"))
+            {
+                // it should be impossible to have both a manifest vcpkg-configuration, and a regular
+                // vcpkg-configuration
+                Checks::check_exit(VCPKG_LINE_INFO, configuration_file.empty());
+                config_in_manifest = true;
+            }
+        }
+
+        if (configuration_file.empty() && !has_builtin_baseline && !config_in_manifest)
+        {
+            msg::println(Color::warning, msgUpdateBaselineNoConfiguration);
+            Checks::exit_success(VCPKG_LINE_INFO);
+        }
```

Comment:
This prevents from inserting a built-in baseline if there's none in the manifest. Is that intended?